### PR TITLE
Ksql 665 remove unsupported operations from the grammar

### PR DESCRIPTION
**Pull request by hjafarpour**
This PR cleans up the grammar. It closed issue #665.
It removes statements like `UNION`, `INTERSECT`, ... that we do  not support in KSQL yet along with features such as `DISTINCT`, `ORDER BY`, ... that we don't have them yet.
Some of the unused features such as `CASE` were not removed since we will have them added soon.

### CODE REVIEW

File: ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java [hunk 1]
```diff
@@ -572,41 +545,6 @@ public Node visitSingleGroupingSet(SqlBaseParser.SingleGroupingSetContext contex
     );
   }
 
-  @Override
-  public Node visitMultipleGroupingSets(SqlBaseParser.MultipleGroupingSetsContext context) {
```
Comment (apurvam) [inline ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java]: Where did this code come from? Did we just copy the grammar and the parser code wholesale from somewhere? I thought we had just copied the grammar.

File: ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/SqlFormatterQueryRewrite.java [hunk 2]
```diff
@@ -119,36 +113,8 @@ protected Void visitExpression(Expression node, Integer indent) {
 
     @Override
     protected Void visitQuery(Query node, Integer indent) {
-      if (node.getWith().isPresent()) {
```
Comment (apurvam) [inline ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/SqlFormatterQueryRewrite.java]: This seems exactly the same as the `SQLFormatter` method. How are these classes different?

File: ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java (hunk 1 above)
Comment (hjafarpour) [reply ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java]: Most of these are from the unused statements in presto grammar.

File: ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/SqlFormatterQueryRewrite.java (hunk 2 above)
Comment (hjafarpour) [reply ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/SqlFormatterQueryRewrite.java]: They were duplicate. removed this one.

File: ksql-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4 [hunk 3]
```diff
@@ -355,7 +328,6 @@ whenClause
 over
     : OVER '('
         (PARTITION BY partition+=expression (',' partition+=expression)*)?
```
Comment (satybald) [inline ksql-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4]: does KSQL support window functions?

File: ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java (hunk 1 above)
Comment (satybald) [reply ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java]: @apurvam writing you SQL parser from scratch is quite a tedious job. Usually, DB folks prefer to reuse existing one rather write your own. 

https://crate.io/how-we-chose-the-facebook-presto-sql-parser-for-crate-data/

File: ksql-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4 (hunk 3 above)
Comment (satybald) [reply ksql-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4]: @hjafarpour @apurvam

**Pull request merged**